Code/Engine/Core/System/Implementation/android/InputDevice_android.inl

```diff
@@ -0,0 +1,193 @@
+  RegisterInputSlot(ezInputSlot_TouchPoint5_PositionY, "Touchpoint 5 Position Y", ezInputSlotFlags::IsTouchPosition);
+
+  RegisterInputSlot(ezInputSlot_TouchPoint6, "Touchpoint 6", ezInputSlotFlags::IsTouchPoint);
+  RegisterInputSlot(ezInputSlot_TouchPoint6_PositionX, "Touchpoint 6 Position X", ezInputSlotFlags::IsTouchPosition);
+  RegisterInputSlot(ezInputSlot_TouchPoint6_PositionY, "Touchpoint 6 Position Y", ezInputSlotFlags::IsTouchPosition);
+
+  RegisterInputSlot(ezInputSlot_TouchPoint7, "Touchpoint 7", ezInputSlotFlags::IsTouchPoint);
+  RegisterInputSlot(ezInputSlot_TouchPoint7_PositionX, "Touchpoint 7 Position X", ezInputSlotFlags::IsTouchPosition);
+  RegisterInputSlot(ezInputSlot_TouchPoint7_PositionY, "Touchpoint 7 Position Y", ezInputSlotFlags::IsTouchPosition);
+
+  RegisterInputSlot(ezInputSlot_TouchPoint8, "Touchpoint 8", ezInputSlotFlags::IsTouchPoint);
+  RegisterInputSlot(ezInputSlot_TouchPoint8_PositionX, "Touchpoint 8 Position X", ezInputSlotFlags::IsTouchPosition);
+  RegisterInputSlot(ezInputSlot_TouchPoint8_PositionY, "Touchpoint 8 Position Y", ezInputSlotFlags::IsTouchPosition);
+
+  RegisterInputSlot(ezInputSlot_TouchPoint9, "Touchpoint 9", ezInputSlotFlags::IsTouchPoint);
+  RegisterInputSlot(ezInputSlot_TouchPoint9_PositionX, "Touchpoint 9 Position X", ezInputSlotFlags::IsTouchPosition);
+  RegisterInputSlot(ezInputSlot_TouchPoint9_PositionY, "Touchpoint 9 Position Y", ezInputSlotFlags::IsTouchPosition);
+}
+
+void ezStandardInputDevice::ResetInputSlotValues()
+{
+  for (int id = 0; id < 10; ++id)
+  {
+    // We can't reset the position inside AndroidHandleInput as we want the position to be valid when lifting a finger. Thus, we clear the position here after the update has been performed.
+    if (m_InputSlotValues[ezInputManager::GetInputSlotTouchPoint(id)] == 0)
+    {
+      m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionX(id)] = 0;
+      m_InputSlotValues[ezInputManager::GetInputSlotTouchPointPositionY(id)] = 0;
+    }
+  }
+}
+
+void ezStandardInputDevice::AndroidInputEventHandler(ezAndroidInputEvent& event)
+{
+  event.m_iReturn = AndroidHandleInput(event.m_pEvent);
+  SUPER::UpdateInputSlotValues();
+}
+
+ezInt32 ezStandardInputDevice::AndroidHandleInput(AInputEvent* pEvent)
+{
+  // #TODO_ANDROID Only touchscreen input is implemented right now.
+  const ezInt32 iEventType = AInputEvent_getType(pEvent);
+  const ezInt32 iEventSource = AInputEvent_getSource(pEvent);
+  const ezUInt32 uiAction = (ezUInt32)AMotionEvent_getAction(pEvent);
+  const ezInt32 iKeyCode = AKeyEvent_getKeyCode(pEvent);
+  const ezInt32 iButtonState = AMotionEvent_getButtonState(pEvent);
+  (void)iKeyCode;
+  (void)iButtonState;
+  DEBUG_LOG("Android INPUT: iEventType: {}, iEventSource: {}, uiAction: {}, iKeyCode: {}, iButtonState: {}", iEventType,
+    iEventSource, uiAction, iKeyCode, iButtonState);
+
+  ezHybridArray<ezScreenInfo, 2> screens;
+  if (ezScreen::EnumerateScreens(screens).Failed() || screens.IsEmpty())
```

Comment:
shouldn't enumerating screens only be done once ?

Reply:
Ah I was just too lazy to implement onResize logic to detect when the screen size changes, adding that now.